Remove deprecated SSL settings and simplify SSL configuration (#213)
logstash-plugins/logstash-input-elasticsearch: docs/index.asciidoc

[id="plugins-{type}s-{plugin}-options"]
==== Elasticsearch Input configuration options

This plugin supports these configuration options plus the <<plugins-{type}s-{plugin}-common-options>> described later.

NOTE: As of version `5.0.0` of this plugin, a number of previously deprecated settings related to SSL have been removed.
Please check out <<plugins-{type}s-{plugin}-obsolete-options>> for details.

> 
> NOTE: As of version `5.0.0` of this plugin, a number of previously deprecated settings related to SSL have been removed.
> Please check out <<plugins-{type}s-{plugin}-obsolete-options>> for details.

Reviewer: ```suggestion
```

Reviewer: Duplicate note

> This plugin supports the following configuration options plus the <<plugins-{type}s-{plugin}-common-options>> and the <<plugins-{type}s-{plugin}-deprecated-options>> described later.

Reviewer: ```suggestion
This plugin supports these configuration options plus the <<plugins-{type}s-{plugin}-common-options>>  described later.
```

Reviewer: This suggestion:
* Removes antiquated language ("the following..." )
* Removes inaccurate info suggesting that we still support deprecated/obsolete settings
* Removes invalid reference so that we won't break the docs builds

empty string authentication will be disabled.

> [id="plugins-{type}s-{plugin}-removed-options"]

Reviewer 2: For the docs update, let's follow the example in https://github.com/logstash-plugins/logstash-output-http/commit/2fae93e26b103cf9e7c0ec0d257be903070805b9#diff-cae5619b3d18ec99c5ccd0a9f6de0c6d3f53343c64692444551a7d29da6863e7

ie, remove all inline references to deprecated settings, and add a dedicated section at the bottom, above "common options", with a link to that section just before the config table.

cc @karenzone

empty string authentication will be disabled.

[id="plugins-{type}s-{plugin}-obsolete-options"]

> ==== HTTP Output Obsolete Configuration Options

Author: Yep, fixed in amended commit. Good catch.
